staging/src/k8s.io/client-go/transport/transport.go

```diff
@@ -119,6 +124,24 @@ func TLSConfigFor(c *Config) (*tls.Config, error) {
 	return tlsConfig, nil
 }
 
+// warnIfExpired checks the NotAfter date of the certificate and compares it to
+// the local system time, logging a warning if the cert is expired.
+func warnIfExpired(c *tls.Certificate) error {
+	now := time.Now()
+	for _, certData := range c.Certificate {
+		parsedCert, err := x509.ParseCertificate(certData)
+		if err != nil {
+			return err
+		}
```

Comment:
I do not think this should error out at all:

`func warnIfExpired(c *tls.Certificate) {`

and just log any `ParseCertificate` error and continue.

Reply:
@enj this could error out if there is a issue with parsing right?

Reply:
> @enj this could error out if there is a issue with parsing right?

No, I think it should just warn along the lines of "I see you are using a client cert but I cannot seem to parse it to determine if it has expired."

I do not want a change in control flow.